lib/rules-generator.js

```diff
@@ -403,7 +416,15 @@ util.methods(Generator, {
       return exp;
 
     case 'var':
+      var valueType = exp.valueType;
       exp = lookupVar(exp);
+      if (valueType) {
```

Comment:
Formally, I think the definition of valueType is a bit confused at the moment.  Possible values are 'Snapshot', 'val', and 'RegExp'.  The intention is to annotate expressions when we have more specific knowledge of the data produced by evaluating the expression.  We use this to coerce Snapshot references to values (exp.val()), coercing .val() results to Booleans (exp == true), and to coerce String constants to RegExp - used in match function - but Bolt does not parse regexp format.

A subsequent PR, replaces adds a notion of TypeExpression, with their own AST representation (allowing for Map<X, Y>).  There may be some chance to unify and make this more concrete.  Note that, where we currently annotate something as String, in many cases it is Snapshot<String>.

This is another area where I can probably improve things my using static typing, and enums in lieu of random string annotations.